src/MainWindow/MessagesTabWidget.cpp

```diff
@@ -0,0 +1,60 @@
+#include "MessagesTabWidget.h"
+
+#include <QGridLayout>
+#include <QTreeWidget>
+
+#include "Compiler/TaskManager.h"
+#include "NewProject/ProjectManager/project_manager.h"
+#include "TextEditor/text_editor_form.h"
+
+namespace FOEDAG {
+
+MessagesTabWidget::MessagesTabWidget(const TaskManager &taskManager)
+    : m_taskManager{taskManager} {
+  auto layout = new QGridLayout();
+  auto treeWidget = new QTreeWidget();
+
+  layout->addWidget(treeWidget);
+  layout->setMargin(0);
+  setLayout(layout);
+
+  treeWidget->setColumnCount(1);
+  treeWidget->setHeaderLabel(tr("Task Messages"));
+
+  auto &reports = m_taskManager.getReportManagerRegistry();
+  for (auto task : m_taskManager.tasks()) {
+    auto taskId = m_taskManager.taskId(task);
+
+    if (auto reportManager = reports.getReportManager(taskId)) {
+      auto taskItem = new QTreeWidgetItem({task->title()});
+      taskItem->setData(0, Qt::UserRole, task->logFileReadPath());
+
+      const auto &msgs = reportManager->getMessages();
```

Comment:
Did you think about adding user-defined log levels to the messages ? Also, I think it's a good idea to let user know the current log file location in this widget.

Reply:
I was thinking about various messages "categories" - error, warning info. Will add support for that in one of following PRs.
We can consider User-configurable stuff separately. It shouldn't result in API change, just some extensions here and there.

As for the log file location, I added it next to the task name in parenthesis (see updated PR description screenshot)